tests/components/shelly/test_sensor.py

```diff
@@ -371,7 +373,7 @@ async def test_block_sensor_unknown_value(
     monkeypatch.setattr(mock_block_device.blocks[block_id], attribute, value)
     mock_block_device.mock_update()
 
-    assert hass.states.get(entity_id).state == STATE_UNKNOWN
+    assert hass.states.get(entity_id).state == value or STATE_UNKNOWN
```

Comment:
You can't do an `or` in a test since it does not validate the expected value, pass the `STATE_UNKNOWN` as the expected value were it is needed

Reply:
The original test is needed to test that None is converted in STATE_UNKNOWN.
If we pass "unknown" from the beginning we don't test the conversion.
this is the reason why I didn't use this test in my first imlementation.

Reply:
Tried to explain this above, you need to have two parameters, one for the block patch value and one for the expected value, the initial value can be removed if you want to make the test shorter, if not you need two parameters.